src/app/Dashboard/ClickableAutomatedAnalysisLabel.tsx

```diff
@@ -0,0 +1,103 @@
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import { getLabelDisplay } from '@app/Recordings/Filters/LabelFilter';
+import { ORANGE_SCORE_THRESHOLD, RED_SCORE_THRESHOLD, RuleEvaluation } from '@app/Shared/Services/Report.service';
+import { Label, LabelProps, Popover } from '@patternfly/react-core';
+import { InfoCircleIcon } from '@patternfly/react-icons';
+import React from 'react';
+
+export interface ClickableAutomatedAnalysisLabelProps {
+    label: RuleEvaluation;
+  isSelected: boolean;
+}
+
+export const ClickableAutomatedAnalysisLabel: React.FunctionComponent<ClickableAutomatedAnalysisLabelProps> = ({label, isSelected}) => {
+  const [isHoveredOrFocused, setIsHoveredOrFocused] = React.useState(false);
+  const [isDescriptionVisible, setIsDescriptionVisible] = React.useState(false);
+
+  const handleHoveredOrFocused = React.useCallback(() => setIsHoveredOrFocused(true), [setIsHoveredOrFocused]);
+  const handleNonHoveredOrFocused = React.useCallback(() => setIsHoveredOrFocused(false), [setIsHoveredOrFocused]);
+
+  const style = React.useMemo(() => {
+    if (isHoveredOrFocused) {
+      const defaultStyle = { cursor: 'pointer', '--pf-c-label__content--before--BorderWidth': '2.5px' };
+      if (isSelected) {
+        return { ...defaultStyle, '--pf-c-label__content--before--BorderColor': '#06c' };
+      }
+      return { ...defaultStyle, '--pf-c-label__content--before--BorderColor': '#8a8d90' };
+    }
+    return {};
+  }, [isSelected, isHoveredOrFocused]);
+
+  return (
+    <Popover 
+        aria-label='automated-analysis-description-popover'        
+        isVisible={isDescriptionVisible}
+        shouldOpen={() => setIsDescriptionVisible(true)}
+        shouldClose={() => setIsDescriptionVisible(false)}
+        zIndex={10000}
+        bodyContent={
+            <div style={{
+                textAlign: 'center',
+                marginLeft: '17%',
+            }}>
+                <p style={{fontWeight: 'bold', fontStyle: 'italic'}}>{label.name}</p>
+                <p style={{fontWeight: 'bold'}}>{label.score == -1 ? 'N/A' : label.score.toFixed(1)}</p>
```

Comment:
Can all these inline `style=` and the `style = React.useMemo` things be extracted into the `app.css` somehow, and given more meaningful names?